### PR TITLE
add start/end date to course, create new views for new and edit actions

### PR DESCRIPTION
**Pull request by kagoromo**

### CODE REVIEW

**Comment by nguyenngocha** [comment]
ok

**Comment by nguyenngocha** [comment]
ok

File: app/controllers/supervisor/courses_controller.rb [hunk 1]
```diff
@@ -2,43 +2,47 @@ class Supervisor::CoursesController < ApplicationController
...
     end
   end
 
   def show
   end
 
-  def edit
-    @course.build_course_subjects @course.subjects
+  def new
+    @course = Course.new
```
Comment (NgengChhengkim) [inline app/controllers/supervisor/courses_controller.rb]: `@course = Course.new` doan nay khong can. vi da dung load_and_authorize_resource

File: app/controllers/supervisor/courses_controller.rb [hunk 2]
```diff
@@ -2,43 +2,47 @@ class Supervisor::CoursesController < ApplicationController
   load_and_authorize_resource
 
   def index
+    @courses = @courses.page(params[:page]).per Settings.courses.per_page
     @course = Course.new
     @course.build_course_subjects @course.subjects
     @search = @courses.ransack params[:q]
     unless params[:q].nil?
-      @courses = @search.result.page(params[:page]).per Settings.courses.per_page
+      @courses = @search.result
```
Comment (NgengChhengkim) [inline app/controllers/supervisor/courses_controller.rb]: tai sao can kiem tra  `params[:q].nil?`

File: app/views/supervisor/courses/_course_form.html.erb [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
-          <%= ff.label :subject_id, ff.object.subject.content %>
-        </div>
-      <% end %>
-  
-      <%= f.submit t("courses.submit"),
-        class: "btn btn-primary col-md-4 col-md-offset-4" %>
+    </div>
+    <div class="form-group">
+      <%= f.label :start_date, t("courses.start_date") %>
+      <%= f.date_field :start_date, class: "form-control", required: true %>
```
Comment (NgengChhengkim) [inline app/views/supervisor/courses/_course_form.html.erb]: co validate trong model chua?

Comment (kagoromo) [inline app/views/supervisor/courses/_course_form.html.erb]: đã thêm ạ

File: app/views/supervisor/courses/_course_form.html.erb [hunk 4]
```diff
@@ -1,34 +1,46 @@
...
+<div class="row">
+  <div class="col-md-6">
+    <div class="form-group">
       <%= f.label :content, t("courses.content") %>
       <%= f.text_field :content, class: "form-control", required: true %>
-      
+    </div>
+    <div class="form-group">
       <%= f.label :description, t("courses.description") %>
       <%= f.text_area :description, class: "form-control", required: true %>
```
Comment (NgengChhengkim) [inline app/views/supervisor/courses/_course_form.html.erb]: `description` co bat buoc nguoi ta phai nhap ko? neu co thi chua co validate trong model

File: app/models/course.rb [hunk 5]
```diff
@@ -10,6 +10,7 @@ class Course < ActiveRecord::Base
   accepts_nested_attributes_for :user_courses, allow_destroy: true
 
   validates :content, presence: true
+  validate :valid_dates
```
Comment (NgengChhengkim) [inline app/models/course.rb]: validates :start_date, presence: true
validates :end_date, presence: true

File: app/models/course.rb [hunk 6]
```diff
@@ -39,4 +40,11 @@ def create_course_activity
     started? ? (create_activities I18n.t("activity.started")) :
       (create_activities I18n.t("activity.finished"))
   end
+
+  def valid_dates
+    return if [end_date.blank?, start_date.blank?].any?
```
Comment (NgengChhengkim) [inline app/models/course.rb]: bo di dong `return if [end_date.blank?, start_date.blank?].any?`

**Pull request merged**